### PR TITLE
Login backend 3

### PR DESCRIPTION
**Pull request by dylan-muszel**
### Summary
Login backend with server calls and server validations. 

### Trello Card
https://trello.com/c/BjZRG2sZ/28-u2-backend-login

### CODE REVIEW

File: app/src/main/java/ar/com/wolox/android/training/model/User.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package ar.com.wolox.android.training.model;
+
+
+public class User {
```
Comment (DemianIac) [inline app/src/main/java/ar/com/wolox/android/training/model/User.java]: This object should be Serializable . You can use  @SerializedName("name") on the fields when the name on the Json is something like "personal_phone" , and then private String personalPhone.

It's not necessary to map all the fields either . You can just create the fields that you need.

File: app/src/main/java/ar/com/wolox/android/training/model/User.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    private String username;
+    private String email;
+    private String password;
+    private String picture;
+    private String cover;
+    private String description;
+    private String location;
+    private String name;
+    private String phone;
+
```
Comment (DemianIac) [inline app/src/main/java/ar/com/wolox/android/training/model/User.java:17]: Constructor must be defined too. I like to create the empty one and with all the fields , but that it's up to you :)

File: app/src/main/java/ar/com/wolox/android/training/ui/errors/ErrorHandler.java [hunk 3]
```diff
@@ -9,8 +9,14 @@ public static String getErrorMessage(ErrorCode errorCode) {
             case INVALID_EMAIL:
                 return "El email ingresado es incorrecto";
```
Comment (DemianIac) [inline app/src/main/java/ar/com/wolox/android/training/ui/errors/ErrorHandler.java:10]: I think that you changed this in other branch. Just  don't forget it! 😛

File: app/src/main/java/ar/com/wolox/android/training/ui/login/LoginFragment.java [hunk 4]
```diff
@@ -42,13 +45,16 @@ public int layout() {
...
     public void init() {
-        mEmail.setText(getPresenter().getUserEmail());
         mLoginBtn.setText(R.string.login_login_btn_text);
         mLoginBtn.setColor(R.color.white);
         mSignupBtn.setText(R.string.login_signup_btn_text);
         mSignupBtn.setTextColor(R.color.white);
         mSignupBtn.setColor(R.color.colorPrimary);
         mTermsAndConditions.setMovementMethod(LinkMovementMethod.getInstance());
+
+        if (getPresenter().isUserLoged()) {
```
Comment (DemianIac) [inline app/src/main/java/ar/com/wolox/android/training/ui/login/LoginFragment.java]: Remember to create the RootActivity for this , like we talked about .

File: app/src/main/java/ar/com/wolox/android/training/ui/login/LoginPresenter.java [hunk 5]
```diff
@@ -20,11 +30,30 @@ public LoginPresenter(UserSession userSession) {
     }
 
     public void login(String email, String password) {
-        mUserSession.setEmail(email);
-        getView().onLoginSuccess();
+        mRetrofitServices.getService(UserService.class).login(email, password).enqueue(new NetworkCallback<List<User>>() {
+            @Override
+            public void onResponseSuccessful(final List<User> user) {
+                if (user.size() > 0) {
```
Comment (DemianIac) [inline app/src/main/java/ar/com/wolox/android/training/ui/login/LoginPresenter.java]: You can use the method isEmpty() on list too if you want

File: app/src/main/java/ar/com/wolox/android/training/utils/UserSession.java [hunk 6]
```diff
@@ -23,16 +18,20 @@ public UserSession(SharedPreferencesManager sharedPreferencesManager) {
...
         // application, but we should add a check in case Android decides to kill the application
         // and return to a state where this isn't initialized.
         if (mEmail == null) {
-            mEmail = mSharedPreferencesManager.get(USER_EMAIL, null);
+            mEmail = mSharedPreferencesManager.get(Extras.UserLogin.EMAIL, null);
         }
         return mEmail;
     }
+
+    public boolean isLoged() {
```
Comment (DemianIac) [inline app/src/main/java/ar/com/wolox/android/training/utils/UserSession.java]: just a typo Logged . Remember that you can change every reference of this with  shift+F6

File: app/src/main/java/ar/com/wolox/android/training/utils/Utils.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
+package ar.com.wolox.android.training.utils;
+
+import android.app.Activity;
+import android.content.Intent;
+
+import ar.com.wolox.android.training.ui.login.LoginActivity;
+
+public class Utils {
```
Comment (DemianIac) [inline app/src/main/java/ar/com/wolox/android/training/utils/Utils.java]: Maybe we can call this UserHelper , UserUtils or SessionUtils . Just to give the message to other developers

File: app/src/main/res/layout/fragment_login.xml [hunk 8]
```diff
@@ -9,6 +9,30 @@
...
+        <ProgressBar
+            android:id="@+id/progressbar"
+            style="?android:attr/progressBarStyleHorizontal"
+            android:visibility="invisible"
+            android:layout_width="fill_parent"
+            android:layout_height="wrap_content"
+            android:indeterminate="true"
+            android:max="100"
+            android:layout_below="@+id/framelauout"
+            android:indeterminateTint="#1a09d6"
```
Comment (DemianIac) [inline app/src/main/res/layout/fragment_login.xml]: Hardcode value , you can use the resource file too.

File: app/src/main/res/layout/fragment_login.xml [hunk 9]
```diff
@@ -9,6 +9,30 @@
...
+            android:layout_marginTop="-7dp"/>
```
Comment (DemianIac) [inline app/src/main/res/layout/fragment_login.xml]: Try not to use negative values if possible.

File: app/src/main/res/layout/fragment_login.xml [hunk 10]
```diff
@@ -9,6 +9,30 @@
...
+
+        <ProgressBar
+            android:id="@+id/progressbar"
+            style="?android:attr/progressBarStyleHorizontal"
+            android:visibility="invisible"
+            android:layout_width="fill_parent"
+            android:layout_height="wrap_content"
+            android:indeterminate="true"
+            android:max="100"
+            android:indeterminateTint="#1a09d6"
```
Comment (DemianIac) [inline app/src/main/res/layout/fragment_login.xml]: Same here

**Review by DemianIac — CHANGES_REQUESTED** [review]
Some changes and recommendations

File: app/src/main/java/ar/com/wolox/android/training/ui/errors/ErrorHandler.java (hunk 3 above)
Comment (dylan-muszel) [reply app/src/main/java/ar/com/wolox/android/training/ui/errors/ErrorHandler.java:10]: Yes! Here: https://github.com/wolox-training/dm-android/blob/home-page-3/app/src/main/java/ar/com/wolox/android/training/ui/errors/ErrorHandler.java

**Pull request closed without merging**